h2o-core/src/main/java/hex/ModelMetricsBinomial.java

```diff
@@ -19,15 +19,19 @@
 public class ModelMetricsBinomial extends ModelMetricsSupervised {
   public final AUC2 _auc;
   public final double _logloss;
+  public final double _loglikelihood;
+  public final double _aic;
   public double _mean_per_class_error;
   public final GainsLift _gainsLift;
 
   public ModelMetricsBinomial(Model model, Frame frame, long nobs, double mse, String[] domain,
```

Comment:
I would probably prefer adding one more constructor here and revert all the changes for non-GLM algos that don't support likelihood.

```java
public ModelMetricsBinomial(Model model, Frame frame, long nobs, double mse, String[] domain,
                              double sigma, AUC2 auc, double logloss, double loglikelihood, double aic, GainsLift gainsLift,
                              CustomMetric customMetric) {
....
}

public ModelMetricsBinomial(Model model, Frame frame, long nobs, double mse, String[] domain,
                              double sigma, AUC2 auc, double logloss, GainsLift gainsLift, CustomMetric customMetric) {
    this(model, frame, nobs, mse, domain, sigma, auc, logloss, Double.NaN, Double.NaN, gainsLift, customMetric);
}
```

This way you wouldn't need to modify all the constructors in all the models that use `ModelMetricsBinomial` and you could provide some reasonable values as defaults for loglikelihood and AIC. For example if I add a new algo, I want it to be easy so I don't have to think with which values I should fill fields that the algo doesn't support. Having the defaults in a new constructor makes it easier and more consistent.

I would recommend using some value like `NaN` for the default - if the loglikelihood is really log likelihood (and not negative log likelihood) then `0` is the maximum and that doesn't seem like a good default (e.g. think about sorting by that in the leaderboard etc). (And similarly for AIC.)